gazebo/gui/ApplyWrenchDialog.cc

```diff
@@ -486,14 +486,22 @@ bool ApplyWrenchDialog::SetModel(const std::string &_modelName)
   for (unsigned int i = 0; i < vis->GetChildCount(); ++i)
   {
     rendering::VisualPtr childVis = vis->GetChild(i);
+
+    // Skip all children that aren't links
+    if (childVis->GetType() != rendering::Visual::VT_LINK)
+    {
+      continue;
+    }
+
     std::string linkName = childVis->Name();
 
     // Issue #1553: This is failing to get real links sometimes:
     // uint32_t flags = childVis->GetVisibilityFlags();
     // if (!((flags != GZ_VISIBILITY_ALL) && (flags & GZ_VISIBILITY_GUI)))
     if (linkName.find("_GL_MANIP_") == std::string::npos)
     {
-      std::string unscopedLinkName = linkName.substr(linkName.find("::") + 2);
+      std::string unscopedLinkName = linkName.substr(linkName.rfind("::") + 2);
```

Comment:
nit: with nested <include> tags not getting fixed, it might actually be better to find/rfind the scoped name of the model instead of using only the deliminator.

Reply:
I think why they originally used `std::string::find()` here is because getting the scoped name of the Model object requires having a ModelPtr or more generic BasePtr; however, this doesn't exist in `ApplyWrenchDialog.cc` since `ApplyWrenchDialog::Init(const std::string &_modelName, const std::string &_linkName)` gets the model name and link name as a string. That's why I decided to use `std::string::rfind()`. I'm assuming you're referring to the "GetScopedName()" function, right?

Reply:
GetScopedName() is server side and not really applicable here. I just mean that we can't actually assume '::' is the deliminator between entities because the way sdformat < 11 resolves `<include>` tags in a way that link (unscoped) names might include '::', so you might be cutting off part of the link name if you rfind

Reply:
Just resolved this in the most recent commit